<?xml version="1.0"?>
<launch>
    <arg name="world_file_name"    default="2d_scenario_V4.xml" />
    <arg name="scenario_path" default="$(find uav_path_planning)/benchmarks/scenarios/2d/" />
    <arg name="world_path" default="$(find uav_path_planning)/benchmarks/scenarios/2d/$(arg world_file_name)" />

    <group ns="path_planning">
        <param name="path_to_gazebo_xml" value="$(arg world_path)" />
        <param name="path_to_classifier_csv" value="$(find uav_path_planning)/benchmarks/classification_benchmark.csv" />
        <param name="path_to_waypoints_pickle" value="$(arg scenario_path)/waypoints_V3.p" />
        <param name="tol_wp_local" value=".5" />
        <param name="tol_wp_global" value=".1" />
        <param name="takeoff_altitude" value="0.5" />
        <param name="apf_3D" value="false" />
        <param name="bug_mode_allowed" value="false" />

        <node pkg="uav_path_planning" type="path_logger_node" name="path_logger_node" launch-prefix="xterm -e"/>
        <node pkg="uav_path_planning" type="obstacle_sensor_node" name="obstacle_sensor_node" launch-prefix="xterm -e"/>
        <node pkg="uav_path_planning" type="obstacle_map_node" name="obstacle_map_node"/>
        <node pkg="uav_path_planning" type="local_path_planner_node" name="local_path_planner_node" launch-prefix="xterm -e"/>
        <node pkg="uav_path_planning" type="global_path_node" name="global_path_node" launch-prefix="xterm -e"/>
        <node pkg="uav_path_planning" type="calc_apf.py" name="calc_apf" launch-prefix="xterm -e"/>
    </group>

    <!-- MAVROS, PX4 SITL and Gazebo -->
    <include file="$(find px4)/launch/mavros_posix_sitl.launch">
        <arg name="world" value="$(arg world_path)"/>
    </include>

</launch>
